Extension/src/LanguageServer/settings.ts

```diff
@@ -414,337 +432,253 @@ export class CppSettings extends Settings {
+        const setting = getRawSetting("C_Cpp." + settingName);
+        return setting.default;
     }
 
-    public get vcFormatSpaceRemoveAroundUnaryOperator(): boolean {
-        return super.Section.get<boolean>("vcFormat.space.removeAroundUnaryOperator") === true;
-    }
+    // Returns the value of a setting as a string with proper type validation and checks for valid enum values.
+    private getAsString(settingName: string): string;
+    private getAsString(settingName: string, allowNull: boolean): string | null;
+    private getAsString(settingName: string, allowNull: boolean = false): string | null {
+        const value: any = super.Section.get<any>(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
 
-    public get vcFormatSpaceAroundBinaryOperator(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundBinaryOperator")!;
+        if (setting.enum !== undefined) {
+            if (settingName === "loggingLevel" && isNumber(Number(value)) && Number(value) >= 0) {
+                return value;
+            }
+            if (this.isValidEnum(setting.enum, value)) {
+                return value;
+            }
+            return setting.default;
+        }
+        if (isString(value)) {
+            return value;
+        }
+        return setting.default;
     }
 
-    public get vcFormatSpaceAroundAssignmentOperator(): string {
-        // These strings have default values in package.json, so should never be undefined.
-        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-        return super.Section.get<string>("vcFormat.space.aroundAssignmentOperator")!;
+    // Returns the value of a setting as a number with proper type validation and checks if value falls within the specified range.
+    private getAsNumber(settingName: string): number;
+    private getAsNumber(settingName: string, allowNull: boolean): number | null;
+    private getAsNumber(settingName: string, allowNull: boolean = false): number | null {
+        const value: any = super.Section.get<any>(settingName);
+        if (allowNull && value === null) {
+            return null;
+        }
+        const setting = getRawSetting("C_Cpp." + settingName);
+        // Validates the value is a number and clamps it to the specified range. Allows for undefined maximum or minimum values.
+        if (isNumber(value)) {
+            if (setting.minimum !== undefined && value < setting.minimum) {
+                return setting.minimum;
+            }
+            if (setting.maximum !== undefined && value > setting.maximum) {
+                return setting.maximum;
+            }
+            return value;
+        }
+        return setting.default;
```

Comment:
It seems like would be safter to check for undefined in case that is possible -- I haven't checked if it's actually possible. I saw some of the other helper functions check for `undefined`.

Reply:
Well, actually, I'm not sure...maybe it's supposed to throw an exception? Maybe ask Bob what he thinks.

Reply:
getRawSetting is not allowed to return undefined, so `setting` will be an object.